cvmfs/catalog_mgr.h

```diff
@@ -142,8 +144,12 @@ class RemountListener {
  *   catalog_manager->Init();
  *   catalog_manager->Lookup(<inode>, &<result_entry>);
  */
+template <class CatalogT>
```

Comment:
It would be good to add a line of documentation here: `CatalogT` is either `Catalog` or the mocked catalog (or perhaps at some point a read-writable catalog).